lib/scripting/post-process/helpers.ts

```diff
@@ -17,304 +17,103 @@
-	const stmt = result[0];
-	const otherBlockStmt = result[1];
-	return estree.blockStatement(otherBlockStmt != null ? [stmt, ...otherBlockStmt.body] : [stmt]);
-}
-
-export function methodStmtList(result: [Statement, Statement[]]): Statement[] {
-	const stmt = result[0];
-	const otherStmts = result[1] || [];
-	return [stmt, ...otherStmts];
-}
-
-export function exitStmt(result: [Token, null, Token]): Statement {
-	const type = result[2].type;
-	return type === 'kw_do' || type === 'kw_for' ? estree.breakStatement() : estree.returnStatement(null);
-}
-
-export function exprList1(result: [Expression, null, Token, null, Expression[]]): Expression[] {
-	const firstExpr = result[0];
-	const otherExprs = result[4] || [];
-	return [firstExpr, ...otherExprs];
-}
-
-export function exprList2(result: [Expression]): Expression[] {
-	const expr = result[0];
-	return [expr];
-}
-
-export function leftExpr1(result: [Identifier, null, Expression[], Token, Expression]) {
-	const identifier = result[0];
-	const indexOrParams = result[2];
-	const expr = result[4];
-	return estree.memberExpression(estree.callExpression(identifier, indexOrParams), expr);
-}
-
-export function leftExpr2(result: [Expression, null, Expression[], Expression]): Expression | null {
-	const identifier = result[0];
-	const indexOrParamsDot = result[2];
-	const leftExprTail = result[3];
-	const callExpr = estree.callExpression(identifier, indexOrParamsDot);
-	if (leftExprTail.type === 'Identifier') {
-		return estree.memberExpression(callExpr, leftExprTail);
-	} else if (leftExprTail.type === 'CallExpression') {
-		return estree.callExpression(
-			estree.memberExpression(callExpr, leftExprTail.callee as Identifier),
-			leftExprTail.arguments as Expression[],
-		);
+function ppStatements(node: ESIToken): any {
+	let stmts: Statement[] = [];
+	for (const child of node.children) {
+		if (!Array.isArray(child.estree)) {
+			stmts.push(child.estree);
+		} else {
+			stmts = stmts.concat(...child.estree);
+		}
 	}
-	return null;
+	return stmts;
 }
 
-export function leftExpr3(result: [Identifier, null, Expression[]]): CallExpression {
-	const identifier = result[0];
-	const indexOrParams = result[2];
-	return estree.callExpression(identifier, indexOrParams);
-}
-
-export function leftExprTail1(result: [Identifier, null, Expression[], Token, Expression]): Expression {
-	const identifier = result[0];
-	const indexOrParams = result[2];
-	const expr = result[4];
-	let callExpr = estree.callExpression(identifier, indexOrParams);
-	if (expr.type === 'CallExpression') {
-		callExpr = estree.callExpression(
-			estree.memberExpression(callExpr, expr.callee as Identifier),
-			expr.arguments as Expression[],
-		);
+function ppBlock(node: ESIToken): any {
+	let stmts: Statement[] = [];
+	for (const child of node.children) {
+		if (!Array.isArray(child.estree)) {
+			stmts.push(child.estree);
+		} else {
+			stmts = stmts.concat(...child.estree);
+		}
 	}
```

Comment:
I've seen this a few times now.. Maybe a `flattenStatements()` function could do this for all?

Reply:
agreed. Would like to submit a second PR with optimizations and extra helper functions.